.github/workflows/push-head-images.yaml

```diff
@@ -0,0 +1,84 @@
+name: Build and push head images
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
If this is too frequent, we could do a nightly build as well. We typically get between 0-6 merges a day to main, avg around 3. But as team grows that can go up. Scaling this back to once a day will lower cost of producing it. Or if we want to be crazy with throw a random generator in here, build if `rand.Intn(3) ==0`  😄

Reply:
Sure! Let's gather some feedback from the team and change if needed.

Reply:
I love the random number :D. Let's monitor the cost and see where it lands.